channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -216,7 +216,27 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControls {
                     _clearScheduledTimer();
 
                     try {
-                        return _self.triggerSend(isAsync, null, sendReason || SendRequestReason.ManualFlush);
+                        let result = _self.triggerSend(isAsync, null, sendReason || SendRequestReason.ManualFlush);
+                        
+                        // If triggerSend returns a promise, return it
+                        if (isPromiseLike(result)) {
+                            return result;
```

Comment:
Change to use doAwaitResponse

```ts
// Handles non-promise and always called if the returned promise resolves or rejects
return doAwaitresponse(result, (rsp) => {
...
});
```

Reply:
In the _self.triggerSend the path to "result = _self._sender(payload, async);" still doesn't return a promise because the "_doSend" function never returns a promise -- fix the root cause